### PR TITLE
SDK: Add 'Completions'

### PR DESCRIPTION
**Pull request by peteski22**
## What's changing

This PR adds support for the Completions API in the SDK.

## How to test it

Run the PyTest tests locally in your IDE, or `pants test lumigator/python/mzai/sdk:: -- -s`

## I already...

- [x] added some tests for any new functionality;
- [ ] updated the documentation.

### CODE REVIEW

File: lumigator/python/mzai/backend/backend/api/routes/completions.py [hunk 1]
```diff
@@ -5,17 +5,20 @@
...
+
 
 @router.get("/")
 def list_vendors():
-    return ["openai", "mistral"]
+    return [VENDOR_MISTRAL, VENDOR_OPENAI]
 
 
-@router.post("/mistral")
+@router.post(f"/{VENDOR_MISTRAL}")
```
Comment (javiermtorres) [inline lumigator/python/mzai/backend/backend/api/routes/completions.py:17]: Later on, I'd use a more dynamic dict[vendor_name:str, vendor_service: CompletionService] that is stored persistently.

File: lumigator/python/mzai/sdk/completions.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def get_completion(self, vendor: str, text: str) -> CompletionResponse | None:
+        """Returns completions from the specified vendor for given text (prompt)."""
+
+        # Sanitize the inputs.
+        vendor = vendor.lower()
+        text = text.strip()
+
+        # Attempt to validate vendors using the cache.
+        if vendor not in self.__cached_vendors:
+            raise ValueError(f"vendor '{vendor}' not supported")
```
Comment (javiermtorres) [inline lumigator/python/mzai/sdk/completions.py:37]:
```suggestion
            raise ValueError(f"vendor '{vendor}' not found in cache, try reloading vendors")
```
...or something like it

**Review by javiermtorres — APPROVED** [review]
lgtm

File: lumigator/python/mzai/sdk/completions.py (hunk 2 above)
Comment (peteski22) [reply lumigator/python/mzai/sdk/completions.py]: Good point! I'll update it now.

**Pull request merged**